Foundation/FileManager.swift

```diff
@@ -1818,19 +1817,70 @@ open class FileManager : NSObject {
         }
 
         var statInfo = stat()
+#if os(Windows)
+        let h = path.withCString(encodedAs: UTF16.self) {
+            CreateFileW( /*lpFileName=*/$0,
+                        /*dwDesiredAccess=*/DWORD(0),
+                        /*dwShareMode=*/DWORD(FILE_SHARE_READ),
+                        /*lpSecurityAttributes=*/nil,
+                        /*dwCreationDisposition=*/DWORD(OPEN_EXISTING),
+                        /*dwFlagsAndAttributes=*/DWORD(FILE_FLAG_OPEN_REPARSE_POINT | FILE_FLAG_BACKUP_SEMANTICS),
+                        /*hTemplateFile=*/nil)
+        }
+        if h == INVALID_HANDLE_VALUE {
+            let err = GetLastError()
+            if err == ERROR_FILE_NOT_FOUND {
+                errno = ENOENT
+            }
+            throw _NSErrorWithErrno(errno, reading: true, path: path)
+        }
+        var info: BY_HANDLE_FILE_INFORMATION = BY_HANDLE_FILE_INFORMATION()
+        GetFileInformationByHandle(h, &info)
+        // Group id is always 0 on Windows
+        statInfo.st_gid = 0
+        statInfo.st_atime = info.ftLastAccessTime.time_t
+        statInfo.st_ctime = info.ftCreationTime.time_t
+        statInfo.st_dev = info.dwVolumeSerialNumber
+        // inodes have meaning on FAT/HPFS/NTFS
+        statInfo.st_ino = 0
+        statInfo.st_rdev = info.dwVolumeSerialNumber
+        statInfo.st_mode =
+        // On a symlink to a directory, Windows sets both the REPARSE_POINT and
+        // DIRECTORY attributes. Since Windows doesn't provide S_IFLNK and we
+        // want unix style "symlinks to directories are not directories
+        // themselves, we say symlinks are regular files
+            UInt16((info.dwFileAttributes & DWORD(FILE_ATTRIBUTE_REPARSE_POINT)) != 0
+            ? _S_IFREG
+            : (info.dwFileAttributes & DWORD(FILE_ATTRIBUTE_DIRECTORY)) != 0
```

Comment:
Can you extract the checks into variables please, it makes it easier to read.

Reply:
Done